spaceMissionManager-persistance/src/test/java/cz/muni/fi/Dao/CraftComponentDaoImplTest.java

```diff
@@ -0,0 +1,162 @@
+        entityManager.createQuery("delete from CraftComponent ").executeUpdate();
+        entityManager.getTransaction().commit();
+        entityManager.close();
+    }
+
+//    ADD
+    @Test
+    public void testAddComponent(){
+        EntityManager em = entityManagerFactory.createEntityManager();
+        craftComponentDao.addComponent(orbitalModule);
+        assertThat(em.find(CraftComponent.class, orbitalModule.getId())).isEqualTo(orbitalModule);
+        assertThat(em.createQuery("select c from CraftComponent c", CraftComponent.class).getResultList()).hasSize(1);
+        craftComponentDao.addComponent(heatShield);
+        assertThat(em.find(CraftComponent.class, heatShield.getId())).isEqualTo(heatShield);
+        assertThat(em.createQuery("select c from CraftComponent c", CraftComponent.class).getResultList()).hasSize(2);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testAddComponentWithId(){
+        orbitalModule.setId(1L);
+        craftComponentDao.addComponent(orbitalModule);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testAddComponentWithNullName() throws Exception {
+        orbitalModule.setName(null);
+        craftComponentDao.addComponent(orbitalModule);
+    }
+
+//    FIND ALL
+    public void testFindAllComponents(){
+        EntityManager em = entityManagerFactory.createEntityManager();
+        assertThat(craftComponentDao.findAllComponents()).hasSize(0);
+        em.persist(orbitalModule);
+        em.persist(heatShield);
+        assertThat(craftComponentDao.findAllComponents()).hasSize(2);
+        em.persist(retroRocket);
+        assertThat(craftComponentDao.findAllComponents()).hasSize(3);
+    }
+
+//    FIND BY ID
+    @Test
+	public void testFindComponentById() throws Exception {
+        craftComponentDao.addComponent(heatShield);
+        assertThat(craftComponentDao.findComponentById(heatShield.getId())).isEqualToComparingFieldByField(heatShield);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+	public void testFindComponentByIdWithNullId() throws Exception {
+        craftComponentDao.addComponent(retroRocket);
+        craftComponentDao.findComponentById(null);
+    }
+
+    @Test(expectedExceptions = NoResultException.class)
+	public void testFindNonexistingComponent() throws Exception {
+        craftComponentDao.addComponent(retroRocket);
+        craftComponentDao.findComponentById(retroRocket.getId()+1L);
+    }
+
+//    UPDATE
+    @Test
+    public void testUpdateComponent(){
+        craftComponentDao.addComponent(orbitalModule);
+        orbitalModule.setName("Large orbital module");
+        assertThat(craftComponentDao.findComponentById(orbitalModule.getId())).isEqualToComparingFieldByField(orbitalModule);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testUpdateComponentWithId(){
```

Comment:
id should always be set when updating component, try testing update on component with null id